apps/settings/js/panels/developer_service_workers/developer_service_workers.js

```diff
@@ -0,0 +1,257 @@
+/* -*- Mode: Java; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global LazyLoader */
+/* global uuid */
+
+/**
+ * On B2G we cannot directly show the content of about:* pages with access to
+ * privileged code. We already have an about:serviceworkers chrome page showing
+ * the list of installed service workers on Desktop. Unfortunately we cannot
+ * access it from Gaia. To access this information, we need to query the
+ * platform from the System app. And to show it we chose to add a new panel
+ * inside the existing "Developer" panel in the Settings app. The communication
+ * with the Settings app is done via Inter App Communication API.
+ *
+ * From this panel we will be sending IAC messages to the System app to request
+ * the list of registered service workers and to request actions like updating
+ * or unregistering them. The System app will proxy this IAC messages to the
+ * platform with no modification.
+ *
+ * The high level picture is the following:
+ *
+ *      Settings App
+ *          |
+ *          |- IAC
+ *          |
+ *      System App
+ *          |
+ *          |- Chrome/Content events
+ *          |
+ *        Gecko (ServiceWorkersManager accessed from a B2G dedicated component)
+ */
+
+define(function(require) {
+  'use strict';
+
+  /**
+   * @alias module:developer_service_workers/developer_service_workers
+   * @class DeveloperServiceWorkers
+   * @returns {DeveloperServiceWorkers}
+   */
+  var DeveloperServiceWorkers = function() {
+    this._elements = {};
+  };
+
+  DeveloperServiceWorkers.prototype = {
+    /**
+     * Initialization.
+     *
+     * @access public
+     * @memberOf DeveloperServiceWorkers.prototype
+     * @param  {Object} elements
+     */
+    init: function(elements) {
+      this._elements = elements;
+      this.refresh();
```

Comment:
It would be better calling this function after giving initial values to `listeners` and `serviceWorkersCount`.